google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/project/DefaultAppEngineAssetProvider.java

```diff
@@ -16,32 +16,84 @@
 
 package com.google.cloud.tools.intellij.appengine.project;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Ordering;
+import com.google.common.primitives.Booleans;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.module.Module;
 import com.intellij.openapi.project.Project;
 import com.intellij.openapi.vfs.VirtualFile;
 import com.intellij.packaging.artifacts.Artifact;
-import com.intellij.packaging.artifacts.ArtifactManager;
-import com.intellij.packaging.elements.PackagingElementResolvingContext;
 import com.intellij.packaging.impl.artifacts.ArtifactUtil;
 import com.intellij.psi.PsiManager;
+import com.intellij.psi.search.FilenameIndex;
 import com.intellij.psi.xml.XmlFile;
 
 import org.jetbrains.annotations.NotNull;
 import org.jetbrains.annotations.Nullable;
 
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
 public class DefaultAppEngineAssetProvider extends AppEngineAssetProvider {
+  private static final Logger logger = Logger.getInstance(DefaultAppEngineAssetProvider.class);
 
   @Nullable
   @Override
   public XmlFile loadAppEngineStandardWebXml(@NotNull Project project, @NotNull Artifact artifact) {
-    PackagingElementResolvingContext context = ArtifactManager.getInstance(project)
-        .getResolvingContext();
-    VirtualFile descriptorFile = ArtifactUtil
-        .findSourceFileByOutputPath(artifact, "WEB-INF/appengine-web.xml", context);
+    Set<Module> modules
+        = ArtifactUtil.getModulesIncludedInArtifacts(Collections.singletonList(artifact), project);
+
+    return loadAppEngineStandardWebXml(project, modules);
+  }
+
+  @Nullable
+  @Override
+  public XmlFile loadAppEngineStandardWebXml(@NotNull Project project,
+      @NotNull Collection<Module> modules) {
+    List<VirtualFile> appEngineWebXmls = new ArrayList<>();
+
+    for (Module module : modules) {
+      appEngineWebXmls.addAll(FilenameIndex.getVirtualFilesByName(
+          project, "appengine-web.xml", module.getModuleContentScope()));
+    }
 
-    if (descriptorFile != null) {
-      return (XmlFile) PsiManager.getInstance(project).findFile(descriptorFile);
+    if (appEngineWebXmls.size() > 1) {
+      logger.warn(appEngineWebXmls.size() + " appengine-web.xml files were found. "
```

Comment:
It could be useful here to in the log.warn the list of appengine-web.xmls that are found and the one that is going to be used.